src/WebJobs.Script.WebHost/Diagnostics/Extensions/ScriptHostServiceLoggerExtension.cs

```diff
@@ -330,17 +330,33 @@ public static void ScriptHostServiceRestartCanceledByRuntime(this ILogger logger
             _scriptHostServiceRestartCanceledByRuntime(logger, null);
         }
 
-        public static void LogHostInitializationSettings(this ILogger logger, string functionWorkerRuntime, string functionExtensionVersion, string siteExtensionDirectory, bool inStandbyMode)
+        public static void LogHostInitializationSettings(this ILogger logger, string originalFunctionWorkerRuntime, string functionWorkerRuntime,
+            string originalFunctionWorkerRuntimeVersion, string functionsWorkerRuntimeVersion, string functionExtensionVersion, string currentDirectory,
+            bool inStandbyMode, bool hasBeenSpecialized, bool usePlaceholderDotNetIsolated, string websiteSku, string featureFlags,
+            IDictionary<string, string> hostingConfig)
         {
-            var hostInitializationSettings = $@"{{
-    ""HostInitializationSettings"": {{
-        ""functionsWorkerRuntime"": ""{functionWorkerRuntime}"",
-        ""functionsExtensionVersion"": ""{functionExtensionVersion}"",
-        ""siteExtensionDirectory"": ""{siteExtensionDirectory}"",
-        ""inStandbyMode"": {inStandbyMode}
-    }}
-}}";
-            _logHostInitializationSettings(logger, hostInitializationSettings, null);
+            // This is a dump of values for telemetry right now, but eventually we will refactor this
+            // into a proper Options object
+            var initializationSettings = new
+            {
+                OriginalFunctionWorkerRuntime = originalFunctionWorkerRuntime,
+                FunctionsWorkerRuntime = functionWorkerRuntime,
+                OriginalFunctionWorkerRuntimeVersion = originalFunctionWorkerRuntimeVersion,
+                FunctionsWorkerRuntimeVersion = functionsWorkerRuntimeVersion,
+                FunctionsExtensionVesion = functionExtensionVersion,
+                HostDirectory = currentDirectory,
```

Comment:
Note i changed the name of this from `SiteExtensionDirectory` to `HostDirectory` as I realized this wouldn't make sense anywhere but Windows App Service (as it's the only thing that uses a Site Extension).